include/eve/concept/simd_allocator.hpp

```diff
@@ -0,0 +1,20 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Contributors & Maintainers
+  SPDX-License-Identifier: MIT
+*/
+//==================================================================================================
+#pragma once
+
+#include <concepts>
+
+namespace eve
+{
+  template<typename T>
+  concept simd_allocator = requires(T& t)
+  {
+    { t.allocate_aligned(0,1)        } -> std::same_as<void*>;
```

Comment:
std::byte* ? :P

Reply:
void* is more easy for people to write allocator from as most malloc look-a-like returns void*